File: Python/String-1/extra_front.py

# given a string, return a new string made of 3 copies of the first 2 chars of the original string
# the string may be any length
# if there are fewer than 2 chars, use whatever is there
def extra_front(str):
    s = ''

    if len(str) < 2:
        s = str + str + str
    else:
        s = str[:2] + str[:2] + str[:2]

    return s 

print(extra_front('Hello'))
print(extra_front('ab'))
print(extra_front('H'))

File: JavaScript/String-1/extraFront.js

// given a string, return a new string made of 3 copies of the first 2 chars of the original string
// the string may be any length
// if there are fewer than 2 chars, use whatever is there
const extraFront = (str) => {
    let newStr = '';

    if (str.length < 2) {
        newStr = str.substring(0, str.length) + str.substring(0, str.length) + str.substring(0, str.length);
    } else {
        newStr = str.substring(0, 2) + str.substring(0, 2) + str.substring(0, 2);
    }

    return newStr;
};

console.log(extraFront('Hello'));
console.log(extraFront('ab'));
console.log(extraFront('H'));
